src/core/nodes/FelNode.jl

```diff
@@ -98,3 +98,55 @@ function mixed_type_equilibrium_message(
     end
     return out_mess
 end
+
+""" 
+    function copy_tree(root::FelNode, shallow_copy=false)
+    
+Returns a untangled copy of the a tree. Optionally, the flag `shallow_copy` can be used to obtained a copy of the tree with only the names and branchlengths.
+"""
+function copy_tree(root::FelNode, shallow_copy=false)
+
+    root_copy = FelNode(root.branchlength, root.name)
+    stack = [(root, root_copy)]
+
+    while !isempty(stack)
+        node, node_copy = pop!(stack)
+
+        if !shallow_copy
```

Comment:
Note for self: copy_tree(tree) won't work on a tree with `LazyPartition`s. Maybe this is just due to the fact that they share the global `memoryblocks` stack, but perhaps I could remedy this by better design (of particularily `copy_partition`).